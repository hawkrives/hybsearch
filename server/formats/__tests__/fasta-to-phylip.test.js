/* eslint-env jest */
require('jest-specific-snapshot')

const fastaToPhylip = require('../fasta-to-phylip')
const hashNames = require('../fasta/hash-names')
const fs = require('fs')
const path = require('path')

const base = path.join(__dirname, '..', '..', '..', 'data', '__supporting__')
const whitelist = ['emydura-short.gb']
const files = fs
	.readdirSync(base)
	.filter(name => !name.startsWith('.') && !name.startsWith('_'))
	.filter(name => name.endsWith('.gb'))
	.filter(name => whitelist.includes(name))

for (const file of files) {
	test(file, () => {
		const content = fs.readFileSync(path.join(base, file, 'aligned-fasta'), 'utf-8')
		const tree = fastaToPhylip(content, hashNames(content))
		expect(tree).toMatchSpecificSnapshot(path.join('.', '__snapshots__', 'fasta-to-phylip.' + file + '.hybsnap'))
	})
}
