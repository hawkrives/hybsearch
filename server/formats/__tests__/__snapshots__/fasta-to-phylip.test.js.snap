// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`emydura-short.fasta 1`] = `
"8 843
b50c0d0cbx1 -ggaacaataaattatcacctcaaaagacacataaaaccacaggaaccccta-ctcaaccataaaaaacacaaatccactattaaaaatcattaacaacacctttattgatctccccaccccatctaacatctctactttatgaaactttggttcattactaggggcatgcctcattctacaactagccacaggaatcttcttagctatacactactcatctgatatttccatagcattctcatcaatctcccacatccaacgagacgttcaatatggctgactaattcgaaatatacacgctaacggtgcttcattattctttatatgcatttacctccatattggacgaggaatctactacggttcatacctttacaagaaaacctgaaacactggagtaatattactactcctagtcatagccactgcattcgtgggctacgtactaccatgaggacaaatatcattctgaggagctacagtaatcaccaatctcctatcagccattccatatgtaggccctacacttgtagaatgaatttgaggggggttctccgtagataacgccacccttacccgattctttacattccacttcctaatcccattcgccatcctaggaataactatactacatctactactactacatgaaacaggatcaaacaacccaacaggactaaactcaaactgtgacaaaatcccattccacccatatttctcctataaagatctactaggccttattctaataataacactcctactcacccttaccctattctaccctaacctactaggagatccagacaactttacaccagccaacccactaaccacc
a13630e35x1 ggaacaatcaattattaccccacaaagacacataaaaccacaggaacccctactccaaccataaaaaacacaaatccactattaaaaatcattaacaacaccttcgttgatcttcccaccccatccaacatctccgctttatgaaacttcggatcattactaggagcatgcctcattctacaactagccacaggaatcttcttagctatacactactcatccgacatttctacagcattctcatcaatttcccacatccaacgagatgttcaatatggatgacttattcgaaatatacacgccaacggtgcttcattattctttatatgcatctaccttcatattggacgagggatctactacggttcctacctttacaacaaaacctgaaacactggagtaatattactactcctagttatagccactgcattcgtgggctacgtgctaccatgaggccaaatatcattctgaggggctacagtaatcaccaacctattatcagccattccatacgtaggcccaacacttgtagagtggatttgaggagggttctccgtagacaacgctactcttactcgattcttcacatttcactttctaatcccattcgccatcctaggaataaccctactacacctcctacttctacacgaaacaggatcaaacaacccaacaggactaaactcaaactgtgacaaaatcccattccacccatatttctcgtataaagacctactaggcctcatcctaataattatatgtctactcacccttaccctatttcacccaaacctactaggagatccagacaactttacaccagccaatccactaaccacc
a13630e35x2 ggaacaatcaattattaccccacaaagacacataaaaccacaggaacctctactccaaccataaaaaacacaaatccactattaaaaatcattaacaacaccttcgttgatcttcccaccccatccaacatctccgctttatgaaacttcggatcattactaggagcatgcctcattctacaactagccacaggaatcttcttagctatacactactcatccgacatttctacagcattctcatcaatttcccacatccaacgagatgttcaatatggatgacttattcgaaatatacacgccaacggtgcttcattattctttatatgcatctaccttcatattggacgagggatctactacggttcctacctttacaacaaaacctgaaacactggagtaatattactactcctagttatagccactgcattcgtgggctacgtgctaccatgaggccaaatatcattctgaggggctacagtaatcaccaacctattatcagccattccatacgtaggcccaacacttgtagagtggatttgaggagggttctccgtagacaacgctactcttactcgattcttcacatttcactttctaatcccattcgccatcctaggaataaccctactacacctcctacttctacacgaaacaggatcaaacaacccaacaggactaaactcaaactgtgacaaaatcccattccacccatatttctcgtataaagacctactaggcctcatcctaataattatatgtctactcacccttaccctatttcacccaaacctactaggagatccagacaactttacaccagccaatccactaaccacc
defd70034x1 -ggaacaataa--attatccccaaaagacacataaaactacaggaa-cccctactcaaccataaaaaacacaaacccactactaaaaatcattaataatacctttattgatctccccactccacctaacatctccgctttgtgaaactttggatcattactaggggcatgcctcgctctacaactagccacaggaatcttcttagcgatacactactcatccgatatctccatagcattctcatcaatctcccacatccaacgagatgttcaatatggctgactaattcgaaatatacacgccaacggtgcttcattattctttatatgcatttacctccatattggacgaggaatttactacggttcatatctttacaacaaaacctgaaacactggggtaatattactactcctagtcatagccactgcattcgtgggatacgtactaccatgaggacaaatatcattctgaggagctacagtaatcaccaatctcttatcagccattccatacgtaggccctacacttgtagagtgaatctgagggggattctccgtagatagtgctacccttactcgattcttcacattccactttctaatcccattcgctatcctagggataactatactacatctactactgctacatgaaacaggatcaaacaacccaacaggactaaactcaaactgtgacaaaatcccattccacccatatttctcttataaagatctactaggcctcattctgataatcacacttctactcactcttaccctattctaccccaacctactaggagacccagacaactttacaccagccaacccactaaccacc
defd70034x2 -ggaacaataa--attatccccaaaagacacataaaactacaggaa-cccctactcaaccataaaaaacacaaacccactattaaaaatcattaataatacctttattgacctccccactccacctaacatctccgctttgtgaaactttggatcattactaggggcatgcctcgctctacaactagccacaggaatcttcttagcgatacactactcatccgatatctccatagcattctcatcaatctcccacatccaacgagatgttcaatatggctgactaattcgaaatatacacgccaacggtgcttcattattctttatatgcatttacctccatattggacgaggaatttactacggttcatatctttacaacaaaacctgaaacactggggtaatattactactcctagtcatagccactgcattcgtgggatacgtactaccatgaggacaaatatcattctgaggagctacagtaatcaccaatctattatcagccattccatacgtaggccctacacttgtagagtgaatctgagggggattctccgtagatagtgctacccttactcgattcttcacattccactttctaatcccattcgctatcctagggataactatactacatctactattgctacatgaaacaggatcaaacaacccaacaggactaaactcaaactgtgacaaaatcccattccacccatatttctcttataaagatctactaggcctcattctgataatcacacttctactcactcttaccctattctaccccaacctactaggagacccagacaactttacaccagccaacccactaaccacc
b50c0d0cbx2 -ggaacaataaattaccaccccaaaagacacataaaaccacaggaacccctaccccaaccataaaaaacacaaatccactattaaaaatcattaacaataccttcattgacctccccaccccatctaacatctccgctctatgaaactttgggtcattactaggggcatgccttattctacaactagccacaggaatcttcttagctatacactactcatccgacatttccatagcattctcatcaatctcccacatccaacgagacgttcaatatggctgactaattcgaaatatacacgctaacggtgcttcattattctttatatgcatttacctccatattgggcgaggaatttactacggttcatacctttacaagaaaacttgaaacactggagtaatattactactcctagtaatagccactgcattcgtgggctacgtactaccatgaggacaaatatcattctgaggggctacagtaatcactaatcttctatcagccattccatatgtaggccctacacttgtagaatgaatttgaggaggattctccgtagataacgccacccttacccgattctttacattccactttctaatcccattcgccatcctgggaataactatactacatctactactactacatgaaacaggatcaaataacccaacaggactaaactcaaactgtgacaaaatcccattccacccatatttctcttataaagacctactaggccttattctaataatcacactcctactcacccttaccctgttctaccccaacctactaggagacccagacaactttacaccagccaacccactgaccacc
b50c0d0cbx3 -ggaacaataaattacctccccaaaagacacataaaactacaggaacccctacctcaaccataaaaaacacaaacccactattaaaaatcattaataatacctttattgatctccccactccacctaacatctccgctttgtgaaactttggatcattactaggggcatgcctcgctctacaactagccacaggaatcttcttagcgatacactactcatctgatatctccatagcattctcatcaatctcccacatccaacgagatgttcaatatggctgactaattcgaaatatacacgccaacggtgcttcattattctttatatgcatttacctccatattggacgaggaatttactacggttcatacctttacaacaaaacctgaaacactggggtaatattactactcctagtcatagccactgcattcgtgggatacgtactaccatgaggacaaatatcattctgaggagctacagtaatcaccaacctcttatcagccattccatacgtaggccctacacttgtagagtgaatctgagggggattctccgtagatagtgctacccttactcgattcttcacattccactttctaatcccattcgctatcctagggataactatactacatctactactgctacatgaaacaggatcaaacaacccaacaggactaaactcaaactgtgacaaaatcccattccacccatatttctcttataaagatctactaggcctcattctgataatcacacttctactcactcttaccctattctaccccaacctactaggagacccagacaactttacaccagccaacccactaaccacc
0974e9006x1 -ggaacaatcaattattacccacaaagacacataaaaccacaggaacctctactccaaccataaaaaacacaaatccactattaaaaatcattaataacacctttgttgatcttcccaccccatccaatatctccgccttatgaaacttcggatcattactaggggcatgcctcattctacaactagccacaggaattttcttagccatacactactcatccgacatttctacagcattctcatcaatttcccacatccaacgagatgttcaatatggatgacttattcgaaatatacacgccaacggtgcttcattattctttatatgcatctaccttcatattggacgaggaatctattacggttcctacctttacaataaaacctgaaacactggagtaatattactactactagtcatagccactgcattcgtaggctacgtactaccatgaggccaaatatcattctgaggggctacagtaatcaccaacctattatcagccatcccatacgtaggcccaacacttgtagagtgaatttgaggagggttctccgtagacaacgccacccttactcgattcttcacatttcactttctaatcccattcgccatcctaggaataaccctactacacctcctacttctacacgaaacaggatcaaataacccaacaggactaaactcaaactgtgacaaaatcccattccacccatatttctcgtataaagacctactaggactcatcctaataatcatatgtctacttaccctcaccctattttacccaaacctactaggagatccagacaactttacaccagccaatccactaaccact
"
`;
